tests/cases/conformance/types/rest/objectRest.ts

```diff
@@ -36,3 +36,5 @@ let computed = 'b';
 let computed2 = 'a';
 var { [computed]: stillNotGreat, [computed2]: soSo,  ...o } = o;
 ({ [computed]: stillNotGreat, [computed2]: soSo, ...o } = o);
+
+var noContextualType = ({ aNumber = 12, ...notEmptyObject }) => aNumber;
```

Comment:
OK, so I added back the special case. Which makes the two tests you asked for pass.